client/ayon_deadline/addon.py

```diff
@@ -46,47 +48,108 @@ def get_plugin_paths(self):
         # abstract on the parent class
         return {}
 
-    def get_publish_plugin_paths(self, host_name=None):
+    def get_publish_plugin_paths(
+        self,
+        host_name: Optional[str] = None
+    ) -> List[str]:
         publish_dir = os.path.join(DEADLINE_ADDON_ROOT, "plugins", "publish")
         paths = [os.path.join(publish_dir, "global")]
         if host_name:
             paths.append(os.path.join(publish_dir, host_name))
         return paths
 
-    @staticmethod
-    def get_deadline_pools(webservice, auth=None, log=None):
-        """Get pools from Deadline.
+    def get_pools_by_server_name(self, server_name: str) -> List[str]:
+        """Returns dictionary of pools per DL server
+
+        Args:
+            server_name (str): Deadline Server name from Project Settings.
+
+        Returns:
+            Dict[str, List[str]]: {"default": ["pool1", "pool2"]}
+
+        """
+        pools = self._pools_by_server_name.get(server_name)
+        if pools is None:
+            dl_server_info = self.deadline_servers_info[server_name]
+
+            auth = (dl_server_info["default_username"],
```

Comment:
Isn't this always causing to authenticate, even if it is not required/requested?